### PR TITLE
Fix awscli and deps

### PR DESCRIPTION
**Pull request by haampie**
- py-docutils/py-pyyaml specify python 3.11 compat
- bump awscli / py-botocore

### CODE REVIEW

**Comment by alalazo** [comment]
@spackbot run pipeline

**Review by alalazo — DISMISSED** [review]

File: var/spack/repos/builtin/packages/awscli/package.py [hunk 1]
```diff
@@ -13,16 +13,31 @@ class Awscli(PythonPackage):
...
     depends_on("py-botocore@1.13.44", when="@1.16.308", type=("build", "run"))
     depends_on("py-botocore@1.12.169", when="@1.16.179", type=("build", "run"))
-    depends_on("py-docutils@0.10:0.15", type=("build", "run"))
-    depends_on("py-rsa@3.1.2:3.5.0", type=("build", "run"))
-    depends_on("py-s3transfer@0.2.0:0.2", type=("build", "run"))
-    depends_on("py-pyyaml@3.10:5.2", type=("build", "run"))
-    depends_on("py-colorama@0.2.5:0.4.1", type=("build", "run"))
-    depends_on("py-nose", type="test")
+
+    depends_on("py-colorama@0.2.5:0.4", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/awscli/package.py]: Upper bound needs patch version

File: var/spack/repos/builtin/packages/awscli/package.py [hunk 2]
```diff
@@ -13,16 +13,31 @@ class Awscli(PythonPackage):
...
+
+    depends_on("py-docutils@0.10:0.16", when="@1.27", type=("build", "run"))
+    depends_on("py-docutils@0.10:0.15", when="@1.16", type=("build", "run"))
+
+    depends_on("py-pyyaml@3.10:5.4", when="@1.27", type=("build", "run"))
+    depends_on("py-pyyaml@3.10:5.2", when="@1.16", type=("build", "run"))
+
+    depends_on("py-rsa@3.1.2:4.8.0", when="@1.27", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/awscli/package.py]:
```suggestion
    depends_on("py-rsa@3.1.2:4.7", when="@1.27", type=("build", "run"))
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/awscli/package.py [hunk 3]
```diff
@@ -13,16 +13,32 @@ class Awscli(PythonPackage):
...
     depends_on("py-botocore@1.12.169", when="@1.16.179", type=("build", "run"))
-    depends_on("py-docutils@0.10:0.15", type=("build", "run"))
-    depends_on("py-rsa@3.1.2:3.5.0", type=("build", "run"))
-    depends_on("py-s3transfer@0.2.0:0.2", type=("build", "run"))
-    depends_on("py-pyyaml@3.10:5.2", type=("build", "run"))
-    depends_on("py-colorama@0.2.5:0.4.1", type=("build", "run"))
-    depends_on("py-nose", type="test")
-    depends_on("py-mock@1.3.0:", type="test")
+
+    depends_on("py-colorama@0.2.5:0.4.6", when="@1.27", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/awscli/package.py]:
```suggestion
    depends_on("py-colorama@0.2.5:0.4.4", when="@1.27", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/awscli/package.py [hunk 4]
```diff
@@ -13,16 +13,32 @@ class Awscli(PythonPackage):
...
+    depends_on("py-colorama@0.2.5:0.3.9", when="@1.16", type=("build", "run"))
+
+    depends_on("py-docutils@0.10:0.16", when="@1.27", type=("build", "run"))
+    depends_on("py-docutils@0.10:0.15", when="@1.16", type=("build", "run"))
+
+    depends_on("py-pyyaml@3.10:5.4", when="@1.27", type=("build", "run"))
+    depends_on("py-pyyaml@3.10:5.2", when="@1.16", type=("build", "run"))
+
+    depends_on("py-rsa@3.1.2:4.7.0", when="@1.27", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/awscli/package.py]:
```suggestion
    depends_on("py-rsa@3.1.2:4.7", when="@1.27", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/awscli/package.py [hunk 5]
```diff
@@ -13,16 +13,32 @@ class Awscli(PythonPackage):
...
+    depends_on("py-rsa@3.1.2:3.5.0", when="@1.16", type=("build", "run"))
+
+    depends_on("py-s3transfer@0.6.0:0.6", when="@1.27", type=("build", "run"))
+    depends_on("py-s3transfer@0.2.0:0.2", when="@1.16", type=("build", "run"))
+
+    depends_on("py-mock@1.3.0:", when="@1.16", type="test")
+    depends_on("py-nose", when="@1.16", type="test")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/awscli/package.py]: You can remove these test deps, we don't actually run any tests in this package

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Review by adamjstewart — APPROVED** [review]

**Comment by haampie** [comment]
@spackbot rerun pipeline

**Comment by haampie** [comment]
@spackbot run pipeline

**Comment by haampie** [comment]
@spackbot run pipeline

**Comment by haampie** [comment]
@spackbot run pipeline

**Pull request merged**